### PR TITLE
[Testing] Split up current appium tests into separate projects

### LINKED ISSUE #21411: [Testing] Split up current appium tests into separate projects
### Description

This can probably be done in parallel to this issue
https://github.com/dotnet/maui/issues/21410

This will get our CI and cake scripts ready so we should then be able to just plop in the legacy tests.

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change

Split current appium tests into separate projects.
image

### Issues Fixed

Fixes #21411

### CODE REVIEW

File: src/Controls/samples/Controls.Sample.UITests/Controls.Sample.UITests.WinUI/Controls.Sample.UITests.WinUI.csproj [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    <EnableMsixTooling>true</EnableMsixTooling>
+    <Nullable>enable</Nullable>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <!-- We do not want XAML files to be processed as .NET MAUI XAML, but rather WinUI XAML. -->
+    <EnableDefaultMauiItems>false</EnableDefaultMauiItems>
+    <UseRidGraph>true</UseRidGraph>
+    <DefineConstants>$(DefineConstants);WINDOWS</DefineConstants>
+  </PropertyGroup>
+  
+  <PropertyGroup Condition="'$(RuntimeIdentifierOverride)' != ''">
```
Comment (jsuarezruiz) [inline src/Controls/samples/Controls.Sample.UITests/Controls.Sample.UITests.WinUI/Controls.Sample.UITests.WinUI.csproj:23]:
```
15>C:\Users\jsuarez\.nuget\packages\microsoft.windowsappsdk\1.5.240311000\buildTransitive\Microsoft.Build.Msix.Packaging.targets(1515,5): error APPX1101: Payload contains two or more files with the same destination path 'Platform\Windows\Styles\Resources.xbf'. Source files: 
15>C:\Users\jsuarez\.nuget\packages\microsoft.windowsappsdk\1.5.240311000\buildTransitive\Microsoft.Build.Msix.Packaging.targets(1515,5): error APPX1101: C:\Users\jsuarez\Documents\GitHub\maui\src\Core\src\obj\Debug\net8.0-windows10.0.19041.0\Platform\Windows\Styles\Resources.xbf
15>C:\Users\jsuarez\.nuget\packages\microsoft.windowsappsdk\1.5.240311000\buildTransitive\Microsoft.Build.Msix.Packaging.targets(1515,5): error APPX1101: C:\Users\jsuarez\Documents\GitHub\maui\src\Controls\src\Core\obj\Debug\net8.0-windows10.0.19041.0\Platform\Windows\Styles\Resources.xbf
```

@mattleibow I think you've worked on something related, any ideas?

**Comment by jsuarezruiz** [comment]
Split up the appium tests project by platforms:
image

With this, we can stop using the `IgnoreIfPlatforms ` method (ignored the test on the platform, but launching and navigating to the issue which takes a few seconds) since with compilation directives etc. each platform only has the tests that it is going to execute. With these changes, the process to pass the tests on each platform must take less time.

File: src/Controls/tests/Shared.UITests/Tests/Issues/Issue17366.cs [hunk 2]
```diff
@@ -16,11 +17,9 @@ public Issue17366(TestDevice device) : base(device)
 		[Category(UITestCategories.Brush)]
 		public void Issue17366Test()
 		{
-			this.IgnoreIfPlatforms(new TestDevice[] { TestDevice.Android, TestDevice.Mac, TestDevice.Windows },
-					"The bug only happens on iOS; see https://github.com/dotnet/maui/pull/17789");
-
```
Comment (mattleibow) [inline src/Controls/tests/Shared.UITests/Tests/Issues/Issue17366.cs:21]: Should we keep these messages as comments? I see a few have been deleted, and they usually link to issues that we need for more context.

**Comment by jsuarezruiz** [comment]
Times to pass Android and iOS tests in this PR:
image
Compared to main:
image

**Review by BretJohnson — COMMENTED** [review]
The project names seem a bit backwards from what I expected. Why not use `UITests.Android` instead of `Android.UITests`? That way all the UITests are grouped together in Test Explorer. And naming with a platform suffix is consistent with the convention used in Xamarin and MAUI.

**Comment by PureWeen** [comment]
@jsuarezruiz 

Reminder on docs to update
- https://github.com/dotnet/maui/blob/main/.github/DEVELOPMENT.md

**Review by PureWeen — DISMISSED** [review]

**Review by PureWeen — APPROVED** [review]

**Review by mattleibow — APPROVED** [review]

**Pull request merged**